File: framework/web/src/main/java/cn/bestwu/simpleframework/web/IRespEntity.java
package cn.bestwu.simpleframework.web;

import com.fasterxml.jackson.annotation.JsonIgnore;
import java.util.Map;
import org.springframework.http.HttpStatus;

/**
 * 响应实体
 *
 * @author Peter Wu
 */
public interface IRespEntity {

  default Integer getHttpStatusCode() {
    return HttpStatus.OK.value();
  }

  @JsonIgnore
  Map<String, Object> toMap();
}

File: web/src/main/kotlin/top/bettercode/summer/web/IRespEntity.kt
package top.bettercode.summer.web

import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.http.HttpStatus

/**
 * 响应实体
 *
 * @author Peter Wu
 */
interface IRespEntity {
    @get:JsonIgnore
    val httpStatusCode: Int?
        get() = HttpStatus.OK.value()

    val message: String?

    @JsonIgnore
    fun toMap(): Map<String?, Any?>
}
